Design.md

```diff
@@ -32,71 +32,73 @@ Here are the IELE design decisions described from the perspective of differences
+* The `ret` and `revert` instructions take an arbitrary number of register operands. `revert` reverts the contract, but `ret` will return to the instruction position on the top of the local call stack if one exists, restoring the values of local registers and copying the return values into the return value operands of the call instruction. If the local call stack is empty, it returns from the contract.
+* IELE has a `call .. at` and a `staticcall .. at` instruction used for calls to public functions of other contracts (the target account address is a register operand of the instruction, as is the amount of gas to be spent during the call, and the value to send in case of `call .. at`). `call .. at` and `staticcall .. at` take an arbitrary number of arguments and return values in the form of register operands instead of memory ranges. The arguments are copied into local registers corresponding to the formal arguments of the function being called. When the contract returns, the returned values are copied into the return value registers of the call instruction.
+* If the name of the function being called does not correspond to a public function of the contract being called, an exception occurs.
+* In all cases, if a mismatch occurs between the number of arguments or return values of a function and the matching `call` or `ret` instruction, an exception occurs.
+* Because they are no longer needed, we remove the CALLDATA* instructions and the RETURNDATA* instructions.
+* Because of security concerns, we remove the CALLCODE and DELEGATECALL instructions.
 
 ## Arbitrary Precision Words
 
 Unlike EVM, which uses 32-byte unsigned words, IELE has arbitrary-precision signed words.
 
 * Because words are now explicitly signed, the SDIV, SMOD, SLT, and SGT instructions are removed, and DIV, MOD, LT, and GT are replaced with signed versions.
-* An EXPMOD instruction is added to perform exponentiation modulo a particular base, and the MODEXP precompiled contract is removed.
-* SIGNEXTEND is changed to convert an N-byte twos-complement representation of a signed number into its signed value.
-* a TWOS instruction is added to convert a signed integer into an N-byte twos-complement representation of the number.
-* Because integers are arbitrary-precision, the integer operand of BYTE now counts from the least-significant byte instead of the most-significant.
+* An `expmod` instruction is added to perform exponentiation modulo a particular base, and the MODEXP precompiled contract is removed.
+* `sext` (corresponding to EVM's SIGNEXTEND) is changed to convert an N-byte twos-complement representation of a signed number into its signed value.
+* a `twos` instruction is added to convert a signed integer into an N-byte twos-complement representation of the number.
```

Comment:
`twos` is not intuitive either.  Does EVM has a corresponding operation?

Since we'll write specifications at the IELE level, we want people to be able to read the programs naturally, without having to check the syntax descriptions or the semantics.

Reply:
No, this one is new to IELE. I didn't pick the name and I don't have any suggestion (or strong feelings) on how to rename it.